File: backend/src/services/geminiClient.js
/**
 * Google Gemini API 호출 래퍼 — stateless
 * Claude의 동일한 인터페이스를 노출 (callClaude → callGemini)
 */

import { GoogleGenerativeAI } from "@google/generative-ai";

const apiKey = process.env.GEMINI_API_KEY;
const client = apiKey ? new GoogleGenerativeAI(apiKey) : null;

const DEFAULT_MODEL = process.env.GEMINI_MODEL || "gemini-2.5-flash";

/**
 * @param {object} args
 * @param {string} args.system
 * @param {string} args.userText
 * @param {string} [args.model]
 * @param {number} [args.maxTokens=1024]
 * @returns {Promise<string>}
 */
export async function callGemini({ system, userText, model = DEFAULT_MODEL, maxTokens = 1024 }) {
  if (!client) {
    throw new Error("GEMINI_API_KEY not configured");
  }
  const generativeModel = client.getGenerativeModel({
    model,
    systemInstruction: system,
    generationConfig: {
      maxOutputTokens: maxTokens,
      temperature: 0.3,
    },
  });

  const result = await generativeModel.generateContent(userText);
  return result.response.text();
}

export function isGeminiConfigured() {
  return client !== null;
}

File: ai-server/app/llm/gemini_client.py
"""Google Gemini 클라이언트."""
import os

from .base import LLMProvider

try:
    import google.generativeai as genai
    GEMINI_AVAILABLE = True
except ImportError:
    GEMINI_AVAILABLE = False


class GeminiProvider(LLMProvider):
    def __init__(self):
        self.api_key = os.getenv("GEMINI_API_KEY", "").strip()
        self.model_id = os.getenv("GEMINI_MODEL", "gemini-1.5-flash")
        if GEMINI_AVAILABLE and self.api_key:
            genai.configure(api_key=self.api_key)

    async def call(self, system: str, user_text: str, max_tokens: int = 1024) -> str:
        if not self.is_configured():
            raise RuntimeError("Gemini not configured (GEMINI_API_KEY missing or sdk unavailable)")
        model = genai.GenerativeModel(
            model_name=self.model_id,
            system_instruction=system,
            generation_config={
                "max_output_tokens": max_tokens,
                "temperature": 0.3,
            },
        )
        response = await model.generate_content_async(user_text)
        return response.text

    def is_configured(self) -> bool:
        return GEMINI_AVAILABLE and bool(self.api_key)

    @property
    def name(self) -> str:
        return "gemini"
